x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -526,6 +527,20 @@ public Security(Settings settings) {
             runStartupChecks(settings);
             Automatons.updateConfiguration(settings);
         } else {
+            final List<String> remoteClusterCredentialsSettingKeys = RemoteClusterService.REMOTE_CLUSTER_CREDENTIALS.getAllConcreteSettings(
+                settings
+            ).map(Setting::getKey).sorted().toList();
+            if (false == remoteClusterCredentialsSettingKeys.isEmpty()) {
+                throw new IllegalArgumentException(
+                    org.elasticsearch.core.Strings.format(
+                        "Found [%s] remote clusters with credentials [%s]. Security [%s] must be enabled to connect to them. "
+                            + "Please either enable security or remove these settings from the keystore.",
+                        remoteClusterCredentialsSettingKeys.size(),
+                        Strings.collectionToCommaDelimitedString(remoteClusterCredentialsSettingKeys),
+                        XPackSettings.SECURITY_ENABLED.getKey()
+                    )
+                );
+            }
```

Comment:
Validating here instead of using the Setting validator because Setting validator does not work well with secure settings. The keystore is closed after the components are initialized. But the validator runs every time the setting (`xpack.security.enabled`) is accessed and hence runs into the "keystore is already closed" exception.